projects/packages/waf/src/class-waf-runner.php

```diff
@@ -253,30 +258,39 @@ public static function errorHandler( $code, $message, $file, $line ) { // phpcs:
 	/**
 	 * Initializes the WP filesystem and WAF directory structure.
 	 *
+	 * @throws File_System_Exception If filesystem is unavailable.
+	 * @throws Waf_Exception         If the WAF directory cannot be created.
+	 *
 	 * @return void
-	 * @throws \Exception If filesystem is unavailable.
 	 */
 	public static function initialize_filesystem() {
 		if ( ! function_exists( '\\WP_Filesystem' ) ) {
 			require_once ABSPATH . 'wp-admin/includes/file.php';
 		}
 
 		if ( ! \WP_Filesystem() ) {
-			throw new \Exception( 'No filesystem available.' );
+			throw new File_System_Exception( 'No filesystem available.' );
 		}
 
-		self::initialize_waf_directory();
+		try {
+			self::initialize_waf_directory();
+		} catch ( Waf_Exception $e ) {
+			throw $e;
+		}
```

Comment:
```suggestion
		self::initialize_waf_directory();
```

We could also just not have the try-catch block here I think?

Reply:
Resolved via https://github.com/Automattic/jetpack/commit/c1d3effd93c8fd23355781bca6b324383c18d498